### ISSUE TITLE
SQLLogic tests that 'require httpfs' are not run

### ISSUE THREAD

**Issue by Flogex** [duckdb/duckdb#9625]
### What happens?

When running the SQLLogic tests in CI, the `BUILD_HTTPFS` CMake variable is set to 0. If I have not overlooked something, the flag seems only to be set when using regression tests or other test suites that do not include all SQLLogic tests.

Therefore, tests with `require https` are skipped. An indicator of this is that the following tests do not fail in CI even though they use a [CSV file](https://www.data.gouv.fr/fr/datasets/r/6d186965-f41b-41f3-9b23-88241cc6890c) that is no longer available:
- test/sql/copy/csv/test_csv_httpfs_prepared.test 
- test/sql/copy/csv/test_csv_httpfs.test

In these two test scripts, the URL "https://www.data.gouv.fr/fr/datasets/r/6d186965-f41b-41f3-9b23-88241cc6890c" has to be replaced with another URL pointing to the data set. However, I could not figure out which data set this is.

I propose to either link the httpfs extension into DuckDB when running the SQLLogic tests or, to deal with the non-determinism of network connections, to have a separate test run (e.g., as part of the nightly tests) that executes the test scripts that require the httpfs extension.

A more fundamental issue in my opinion is that the tests are skipped silently and without any warning. This applies to other SQLLogic tests that contain a `require` or `require-env` command, too. Therefore, it is hard to find out that tests are skipped. Of course I understand that logging messages whenever a test is skipped could result in a lot of noise, but at least having an option to enable those warnings would be very nice.

In summary:
- SQLLogic tests that `require httpfs` seem to never be executed in CI
- Two tests use a data set from a URL resulting in 404. The URL should be changed to a new one, if available
- SQLLogic tests are skipped silently when precondition specified by `require` and `require-env` are not met. I suggest to emit a warning in this case

### To Reproduce

Not applicable: This is mainly a CI issue

### OS:

macOS 14

### DuckDB Version:

main

### DuckDB Client:

CLI

### Full Name:

Florian Gerlinghoff

### Affiliation:

MotherDuck

### Have you tried this on the latest `main` branch?

I have tested with a main build

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

**Comment by Mytherin** [comment]
There's a specific [Linux HTTPFS run](https://github.com/duckdb/duckdb/actions/runs/6811547891/job/18525234154) that runs HTTPFS tests. I don't think they are being skipped - there are several tests failing right now actually, and there are plenty of assertions being triggered.

Tests are skipped frequently since it's rare you build all extensions at once so emitting warnings when tests are skipped by default seems like a poor development experience. You can usually tell what tests are skipped based on the number of assertions that are triggered. We could perhaps add a `--verbose` flag that would emit these flags that you can optionally set in the CI if desired.

**Comment by Mytherin** [comment]
Hm, yeah on closer inspection I think you are correct. We should adjust the `Linux HTTPFS` test to run all tests that have `require httpfs`. 

> I agree with your point that emitting those warnings can lead to a poor development experience. But is there a better way to find tests that are, unintentionally, skipped? For example, when some environment variable is not set? Looking at the number of assertions is not a good indicator in my opinion when running many tests, because it is hard to tell the difference between, say, 23,000 and 23,500 assertions. Hence, having a --verbose flag would be very helpful in such a case.

I don't think there is any CI run that actually runs *all* tests. I don't think that's feasible either going forward given that we are always adding more extensions, more tests, ... meaning this would primarily generate a lot of noise that is most likely to be ignored and subsequently not really help. What would be a useful piece of feedback is how often tests are run *in total* across all CI runs, so we could detect tests that are not run at all, but not sure how to accomplish that.
